### ISSUE TITLE
Integrate "dart fix" into flutter tool

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#70113]
This issue is to discuss whether/how we should integrate "dart fix" into flutter tool. Two useful integrations I could see are:

1. Make `flutter fix` available to apply available fixes to my current fluter project (this command would probably just pipe through to "dart fix").
2. Prompt users after upgrading to a new flutter version that they may want to run `flutter fix` to apply any available fixes for breaking changes to their flutter projects (maybe we could automatically detect that new fixes are available in the new version by comparing the dartfix files pre and post upgrade).

/cc @Piinks @jonahwilliams @bwilkerson

**Comment by jonahwilliams** [comment]
1. What does `flutter fix` add over `dart fix`? Generally speaking I'd like to avoid adding more dart tool wrappers to the flutter tool if it is only for "branding" - its nonzero effort to maintain and we're already strapped for time.

2. Generally speaking we should only log error messages if they are actionable - if we at least know they have fixes that can be applied that could be good enough, but this information should also be surfaced through IDEs too.

 ( What happens if someone temporarily upgrades to master/dev/beta, applies fixes, and then wants to go back to master )

**Comment by goderbauer** [comment]
> but this information should also be surfaced through IDEs too.

The fixes do already show up as quick fixes in the IDE.

**Comment by bwilkerson** [comment]
> What happens if someone temporarily upgrades to master/dev/beta, applies fixes, and then wants to go back to master

We aren't currently planning on supporting any form of downgrade. Users would need to use their source code control system to revert the changes.

**Comment by jonahwilliams** [comment]
Seems reasonable!

**Comment by Piinks** [comment]
Re: prompting users to run `flutter/dart fix` when appropriate, could this be determined by running the analyzer and seeing if there is attention needed that way?

**Comment by devoncarew** [comment]
We could add something like `dart fix --dry-run --fix package:flutter` or similar, which could indicate whether there were any package flutter fixes that would apply to the current codebase. That would take ~10sec to run, so not an instant result.

**Comment by jonahwilliams** [comment]
Do we have a way to check if there were new dart fix metadata files added? We could weaken it a bit to detecting potential fixes and directing users to `dart fix`

**Comment by bwilkerson** [comment]
> Do we have a way to check if there were new dart fix metadata files added?

The current design is to have a single file per package, but the tool could easily compare the length of the file or an MD5 hash both before and after the upgrade to determine whether there are any new fixes that users might want to run.

**Comment by jonahwilliams** [comment]
I actually think either of those approaches (running analyzer or checking fixes) will not be feasible on upgrade, since we don't require that command is run in the context of a flutter project.

It would have to be a per-project check of some kind

**Comment by bwilkerson** [comment]
I was thinking of a message more like:

Some APIs have been modified as part of this upgrade. Running `dart fix` in each of your packages might help you migrate your code.

**Comment by christopherfujino** [comment]
cc @Piinks & @goderbauer, do we actually need a dedicated `flutter fix` sub-command, or would it be enough to suggest `dart fix` after a fluter upgrade?

**Comment by goderbauer** [comment]
I don't think we really need the dedicated `flutter fix` subcommand and should direct people to `dart fix` instead.
